docs/docs/spec.md

```diff
@@ -244,10 +240,17 @@ exceptions in the subsequent sections:
 
 -   an array component type
 
-    > An array of nullable strings is written `@Nullable String[]`. A variadic
-    > parameter whose type is "array of nullable strings" is written `@Nullable
-    > String...`. Compare this to the syntax for a nullable array of strings,
-    > shown above in the discussion of variadic parameters.
+    > An array of nullable strings is written `@Nullable String[]`. Similarly, a
+    > variadic parameter whose type is "array of nullable strings" is written
+    > `@Nullable String...`.
+    >
+    > Additionally, whenever JSpecify recognizes an annotation at a given
+    > location, it also recognizes an annotation on *an array itself* at that
```

Comment:
This confused me. I thought you were saying that whenever an annotation can go on an array's component type it can also go on the array itself, which is not true...and then I realized it wasn't what you were saying.

How can we clarify? Maybe start with "When JSpecify recognizes a type-use annotation on an array itself (because it is the type of a field, parameter, return type, etc.), that annotation comes directly before the `[]`, `[dimension]`, or `...` token. For example...."

Reply:
I had a flash of insight—or maybe just the _feeling_ that I'd had a flash of insight. You be the judge :)